pkg/auth/auth.go

```diff
@@ -214,24 +215,39 @@ func (u *UserAuth) isAllowed(verb string, itemsToCheck []resource) ([]Action, er
 	return rejectedActions, nil
 }
 
+func uniqVerbs(current []string, new []string) []string {
+	resMap := map[string]bool{}
+	for _, v := range current {
+		if !resMap[v] {
+			resMap[v] = true
+		}
+	}
+	for _, v := range new {
+		if !resMap[v] {
+			resMap[v] = true
+			current = append(current, v)
+		}
+	}
+	return current
```

Comment:
This function modifies the `current` slice already - which might be unexpected by a call-site. Why not keep it purely functional (ie. returning a new slice).

Reply:
slices get copied to the function context so modifying `current` has no effect for the caller: https://play.golang.org/p/NYgKBEaCIk8

In any case, if it's more readable I can create a different variable

Reply:
> slices get copied to the function context so modifying `current` has no effect for the caller: https://play.golang.org/p/NYgKBEaCIk8

Small change: https://play.golang.org/p/Gu_Ta4jM46n showing that the slice is not being copied and that modifying the slice *does* have an effect for the caller: it's just the slice descriptor which is copied.

But what I missed above is that in this case we're always returning a new slice descriptor anyway (`append` creates one), so you're right - it's safe as is :)

Reply:
ah, got it, good to know